src/components/Footer.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect, Suspense } from "react";
+
+interface CarouselItem {
+  image: string; // image
+  text: string; // text info shown on hover
+}
+
+
+const Footer: React.FC = () => {
+  const [hoverIndex, setHoverIndex] = useState<number | null>(null);
+  const [activeText, setActiveText] = useState<string>("");
+  const [Marquee, setMarquee] = useState<any>(null);
+  const [carouselItems, setCarouselItems] = useState<CarouselItem[]>([]);
+
+  useEffect(() => {
+    // Fetch data from the endpoint
+    fetch()
+      .then(response => response.json())
+      .then(data => {
+        // Transform the data to match CarouselItem structure
+        const items = data.map((item: any) => ({
+          image: item.image_url || "user.png",  // Fallback image if img_url is empty
+          text: `${item.name} - ${item.subteam}`
+        }));
```

Comment:
Can you do some data validation here? I prefer that you validate and assert it to a known data structure. So items that are missing mandatory fields, etc. can be filtered out.